File: src/2108-Find-First-Palindromic-String-in-the-Array/find_first_palindromic_string.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    std::string firstPalindrome(std::vector<std::string>& words) {

        for (const std::string& word : words) {
            int left = 0;
            int right = word.length() - 1;

            while (left < right) {
                if (word[left] != word[right]) {
                    break;
                }
                left++;
                right--;
            }

            if (left >= right) {
                return word;
            }
        }

        return "";
    }
};

int main() {
    std::vector<std::string> words = {"abc", "car", "ada", "racecar", "cool"};

    Solution solution;

    std::string output = solution.firstPalindrome(words);
    std::cout << output << std::endl;
    
    return 0;
}

File: src/2108-Find-First-Palindromic-String-in-the-Array/find_first_palindromic_string.py
from typing import List

class Solution:
    def firstPalindrome(self, words: List[str]) -> str:
        for word in words:
            if word == word[::-1]:
                return word
        
        return ""

if __name__ == "__main__":
    words: List[str] = ["abc","car","ada","racecar","cool"]
    
    solution = Solution()
    
    print(solution.firstPalindrome(words))